### PR TITLE
Modify Rule S6096: Common education content (APPSEC-71)

### PR DESCRIPTION
**Pull request by pierre-loup-tristant-sonarsource**
[APPSEC-71](https://sonarsource.atlassian.net/browse/APPSEC-71)

### CODE REVIEW

File: rules/S6096/rationale.adoc [hunk 1]
```diff
@@ -0,0 +1,6 @@
+Zip slip is a special case of path injection. It occurs when an application uses the name of an archive entry to construct a file path and access this file without validating its path first.
+
+This rule will consider all archives as untrusted, assuming that they have been created outside of the application file system.
```
Comment (pedro-oliveira-sonarsource) [inline rules/S6096/rationale.adoc:3]:
```suggestion
This rule will consider all archives untrusted, assuming they have been created outside the application file system.
```

File: rules/S6096/rationale.adoc [hunk 2]
```diff
@@ -0,0 +1,6 @@
...
+
+A user with malicious intent would inject specially crafted values, such as ``++../++`` in the archive entry name, to change the initial intended path. The resulting path would resolve somewhere in the filesystem where the user should not normally have access to.
```
Comment (pedro-oliveira-sonarsource) [inline rules/S6096/rationale.adoc:5]:
```suggestion
A user with malicious intent would inject specially crafted values, such as ``++../++``, in the archive entry name to change the initial intended path. The resulting path would resolve somewhere in the filesystem where the user should not normally have access.
```

**Review by pedro-oliveira-sonarsource — COMMENTED** [review]
Just a minor change to do on the impact section.

File: rules/S6096/impact.adoc [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+
+The files that can be affected are limited by the permission of the process that runs the application. Worst case scenario: the process runs with root privileges on Linux, and therefore any file can be affected.
+
+Below are some real-world scenarios that illustrate the various effects of an
+attacker exploiting the vulnerability.
+
+==== Override arbitrary files
+
+The archive is open by the application to copy entries to the file system. Archive entry names tamper with the location of a file the application is supposed to copy resulting in files being overwritten. The vulnerability is exploited to corrupt files that are critical for the application or for the system to work properly.
+It could result in data being lost or the application being unavailable.
```
[the remark this answers is not in the document]
Comment (pierre-loup-tristant-sonarsource) [reply rules/S6096/impact.adoc:13]: Not sure is the rule cover this attack vector. Let's discuss it.

File: rules/S6096/metadata.json [hunk 4]
```diff
@@ -26,19 +26,18 @@
...
+      "A5",
       "A1"
     ],
     "OWASP Top 10 2021": [
       "A1",
       "A3"
     ],
     "PCI DSS 3.2": [
-      "6.5.1"
+      "6.5.8"
```
[the remark this answers is not in the document]
Comment (pierre-loup-tristant-sonarsource) [reply rules/S6096/metadata.json:40]: We should also fix S2083 then.

File: rules/S6096/impact.adoc (hunk 3 above)
[the remark this answers is not in the document]
Comment (pierre-loup-tristant-sonarsource) [reply rules/S6096/impact.adoc:13]: I created https://sonarsource.atlassian.net/browse/APPSEC-100 to look into this case.

**Review by pedro-oliveira-sonarsource — APPROVED** [review]
LGTM.

**Pull request merged**